### PR TITLE
Spec tests for Bundles.

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: Firestore/Example/Tests/SpecTests/FSTSpecTests.mm [hunk 1]
```diff
@@ -340,6 +352,17 @@ - (void)doUnlisten:(NSArray *)unlistenSpec {
...
 
+- (void)doLoadBundle:(NSString *)bundleJson {
+  const auto &database_info = [self.driver databaseInfo];
+  BundleSerializer bundle_serializer(remote::Serializer(database_info.database_id()));
+  auto data = util::MakeString(bundleJson);
+  auto bundle = absl::make_unique<util::ByteStreamCpp>(
+      absl::make_unique<std::stringstream>(std::stringstream(data)));
+  auto reader = std::make_shared<BundleReader>(std::move(bundle_serializer), std::move(bundle));
+  auto task = std::make_shared<LoadBundleTask>(user_executor_);
+  [self.driver loadBundleWithReader:reader task:task];
```
Comment (var-const) [inline Firestore/Example/Tests/SpecTests/FSTSpecTests.mm:363]: Optional nit: you can move these variables.

**Review by var-const — CHANGES_REQUESTED** [review]
Almost LGTM -- just a couple of nits.

File: Firestore/Example/Tests/SpecTests/FSTSpecTests.mm (hunk 1 above)
Comment (wu-hui) [reply Firestore/Example/Tests/SpecTests/FSTSpecTests.mm]: Done.

File: Firestore/Example/Tests/SpecTests/FSTSyncEngineTestDriver.h [hunk 2]
```diff
@@ -158,6 +160,17 @@ typedef std::unordered_map<auth::User, NSMutableArray<FSTOutstandingWrite *> *,
  */
 - (void)removeUserListenerWithQuery:(const core::Query &)query;
 
+/**
+ * Loads a firestore bundle captured in `reader` into the local storage..
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply Firestore/Example/Tests/SpecTests/FSTSyncEngineTestDriver.h]: Done.

File: Firestore/Example/Tests/SpecTests/FSTSyncEngineTestDriver.h [hunk 3]
```diff
@@ -158,6 +160,17 @@ typedef std::unordered_map<auth::User, NSMutableArray<FSTOutstandingWrite *> *,
...
+ *
+ * Resulting events are captured and made available via the capturedEventsSinceLastCall method.
+ *
+ * @param reader A object to read from the underlying input stream of the bundle.
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply Firestore/Example/Tests/SpecTests/FSTSyncEngineTestDriver.h]: Done.

File: Firestore/Example/Tests/SpecTests/FSTSyncEngineTestDriver.h [hunk 4]
```diff
@@ -328,6 +341,11 @@ typedef std::unordered_map<auth::User, NSMutableArray<FSTOutstandingWrite *> *,
  */
 @property(nonatomic, readonly) int waitForPendingWritesEvents;
 
+/**
+ * The number of waitForPendingWrites events that have been received.
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply Firestore/Example/Tests/SpecTests/FSTSyncEngineTestDriver.h]: Fixed.

File: Firestore/Example/Tests/SpecTests/FSTSyncEngineTestDriver.mm [hunk 5]
```diff
@@ -476,6 +484,11 @@ - (void)removeUserListenerWithQuery:(const Query &)query {
   }
 }
 
+- (void)loadBundleWithReader:(std::shared_ptr<BundleReader>)reader
+                        task:(std::shared_ptr<LoadBundleTask>)task {
+  _workerQueue->EnqueueBlocking([=] { _syncEngine->LoadBundle(reader, task); });
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply Firestore/Example/Tests/SpecTests/FSTSyncEngineTestDriver.mm]: Done.

**Review by var-const — APPROVED** [review]

**Pull request merged**